src/public-api-types.ts

```diff
@@ -0,0 +1,126 @@
+/**
+ * Types exposed through the public API
+ *
+ * @note I wanted to use the types in the *-plugin.ts files and just assemble them here,
```

Comment:
@dbismut I wasn't sure about this. Do you think there is a way to make these modular?

Reply:
I'm not sure what you mean by "making these modular" 😅? To avoid duplication with types declared in `*-plugin.ts` files you mean?

Reply:
> To avoid duplication with types declared in *-plugin.ts files you mean?

@dbismut yeah exactly. Though having slept on it, I think it'd be way too DRY 😄

Reply:
We could declare the public types in the new file only once and reference the types from the plugins that's fine by me!

Reply:
I just had another look at `*-plugin.ts` files and I think their types aren't meant to be exposed publicly. Like [this one](https://github.com/gsimone/leva/blob/16b84b6242f1caf3d7ba74a457d27b329cbea85a/src/components/Number/number-plugin.ts#L7) from `number-plugin` doesn't include the `number` type while [this public type](https://github.com/AriaMinaei/leva/blob/adfdf697f0ae4830c549e8d7d855bebf6c37325d/src/public-api-types.ts#L10) does. So maybe it's good having the private and public types be separate?

Reply:
Actually some types are indeed public and I think this would prone some errors. So now public input types are only declared in `public-api-types`.